### ISSUE TITLE
Simplify `handleMaybeM` in `hls-plugin-api`

### ISSUE THREAD

**Issue by July541** [haskell/haskell-language-server#3563]
Currently, we have many use cases for `handleMaybeM` like the following, and it gradually being our regular mode of running action. I propose that can simplify it and just pass the action type itself. Like `handleMaybeM' TypeCheck nfp` to get the same effect as `handleMaybeM`.

<img width="768" alt="image" src="https://user-images.githubusercontent.com/16057632/232804864-a58feb3f-c147-4fe0-b466-0b8b4f7e91cd.png">

**Comment by michaelpj** [comment]
I agree, it would be nice to refactor this stuff, it's very verbose and everywhere. I wonder if we should actually give plugins a proper monad to run in rather than `Either String _` or whatever they currently have :thinking:

**Comment by July541** [comment]
`Either String _` is acceptable, but a new design monad may be more suitable for our application.
